services/provider/server/server.go

```diff
@@ -633,24 +638,26 @@ func (s *OCSProviderServer) GetStorageClaimConfig(ctx context.Context, req *pb.S
 				return nil, status.Errorf(codes.Internal, "failed to get %s cephFilesystemSubVolumeGroup. %v", cephRes.Name, err)
 			}
 
-			nodeCephClientSecret, _, err := s.getCephClientInformation(ctx, cephRes.CephClients["node"])
+			_, _, err = s.getCephClientInformation(ctx, cephRes.CephClients["node"])
 			if err != nil {
 				return nil, status.Error(codes.Internal, err.Error())
 			}
 
-			provisionerCephClientSecret, _, err := s.getCephClientInformation(ctx, cephRes.CephClients["provisioner"])
+			_, _, err = s.getCephClientInformation(ctx, cephRes.CephClients["provisioner"])
 			if err != nil {
 				return nil, status.Error(codes.Internal, err.Error())
 			}
 
+			provisionerSecretName := storageClaimCephCsiSecretName("provisioner", storageRequestHash)
+			nodeSecretName := storageClaimCephCsiSecretName("node", storageRequestHash)
 			storageClassName = "cephfs"
 			storageClassData["clusterID"] = getSubVolumeGroupClusterID(subVolumeGroup)
 			storageClassData["subvolumegroupname"] = subVolumeGroup.Name
 			storageClassData["fsName"] = subVolumeGroup.Spec.FilesystemName
 			storageClassData["pool"] = subVolumeGroup.GetLabels()[v1alpha1.CephFileSystemDataPoolLabel]
-			storageClassData["csi.storage.k8s.io/provisioner-secret-name"] = provisionerCephClientSecret
-			storageClassData["csi.storage.k8s.io/node-stage-secret-name"] = nodeCephClientSecret
-			storageClassData["csi.storage.k8s.io/controller-expand-secret-name"] = provisionerCephClientSecret
+			storageClassData["csi.storage.k8s.io/provisioner-secret-name"] = provisionerSecretName
+			storageClassData["csi.storage.k8s.io/node-stage-secret-name"] = nodeSecretName
+			storageClassData["csi.storage.k8s.io/controller-expand-secret-name"] = provisionerSecretName
```

Comment:
I don't see you appending the storage class. I can see something happens after the switch but the does not make sense. You should create it inside the case for that storage class.

Reply:
done.